src/scenes/raylib_game_view.cpp

```diff
@@ -52,81 +68,142 @@ void meow::RaylibGameView::setup_background() {
-    m_usernames_box.draw();
-    m_stats.draw();
-    if (m_stats.m_borders.CheckCollision(raylib::Mouse::GetPosition()) &&
-        raylib::Mouse::IsButtonDown(MOUSE_BUTTON_LEFT)) {
-        m_stats.m_borders.x += raylib::Mouse::GetDelta().x;
-        m_stats.m_borders.y += raylib::Mouse::GetDelta().y;
-    }
-    if (m_show_chat) {
-        if (m_text_chat.get_border().CheckCollision(raylib::Mouse::GetPosition()) &&
-            raylib::Mouse::IsButtonDown(MOUSE_BUTTON_LEFT)) {
-            m_text_chat.position += raylib::Mouse::GetDelta();
+void meow::RaylibGameView::draw() {
+    Overload active_display_visitor = {
+        [this](GameplayObjs *objs) {
+            objs->board.draw(m_window->GetFrameTime());
+            objs->player_hand.draw_cards(m_window->GetFrameTime());
+            objs->usernames_box.draw();
+            if (m_show_chat) {
+                objs->text_chat.draw(*m_client);
+            }
+            objs->stats.draw();  // TODO: move to separate object
+            draw_active_display_selector();
+            GuiCardSpan::draw_inspected_card(m_window->GetWidth(), m_window->GetHeight());
+
+            if (GuiButton({0, 0, 40, 40}, "-") && objs->player_hand.card_count() > 0) {
+                objs->player_hand.remove_card();
+            }
+            if (GuiButton({40, 0, 40, 40}, "+") && objs->player_hand.card_count() < 10) {
+                const auto random_index =
+                    random_integer<std::size_t>(0, m_card_image_paths.size() - 1);
+                objs->player_hand.add_card(m_card_image_paths[random_index].c_str());
+            }
+            if (objs->player_hand.somethind_inspected()) {
+                m_blur.Draw();
+            }
+        },
+        [this](PauseMenuObjs *objs) {
+            m_blur.Draw();
+            draw_pause_menu();
+        },
+        [](auto...) { throw std::runtime_error("unhandled type(s) in active display!"); }
+    };
+
+    // FIXME
+    if (auto action = m_client->receive_action(); action) {
+        if (random_integer(0, 1) == 0) {
+            m_gameplay_objects.board.m_kitten_cards.add_card(action->card_filename);
+        } else {
+            m_gameplay_objects.board.m_opponent_cards.add_card(action->card_filename);
         }
-        m_text_chat.draw(*m_client);
     }
 
-    if (!m_should_draw_pause && GuiButton({0, 0, 40, 40}, "-") && m_player_hand.card_count() > 0) {
-        m_player_hand.remove_card();
-    }
-    if (!m_should_draw_pause && GuiButton({40, 0, 40, 40}, "+") &&
-        m_player_hand.card_count() < 10) {
-        const auto random_index = random_integer<std::size_t>(0, m_card_image_paths.size() - 1);
-        m_player_hand.add_card(m_card_image_paths[random_index].c_str());
+    if (auto chat_message = m_client->receive_chat_message(); chat_message) {
+        std::cout << "received from " << chat_message->sender_player << ": "
+                  << chat_message->message << '\n';
```

Comment:
FYI: it is much more convenient to use special logging libraries such as https://github.com/gabime/spdlog